modules/streamer/src/main/StreamerPager.scala

```diff
@@ -9,72 +9,106 @@ import lila.user.{ User, UserRepo }
-          else
-            $doc(
-              "approval.granted" -> true,
-              "listed"           -> Streamer.Listed(true),
-              "_id" $nin live.streams.map(_.streamer.id)
-            )
+  def nextRequestId: Fu[Option[Streamer.Id]] = coll.primitiveOne[Streamer.Id](
+    $doc("approval.requested" -> true, "approval.ignored" -> false),
+    $sort asc "updatedAt",
+    "_id"
+  )
+
+  private def notLive(live: LiveStreams, forUser: Option[UserId]): AdapterLike[Streamer.WithContext] =
+    new AdapterLike[Streamer.WithContext]:
 
-        def nbResults: Fu[Int] =
-          if (approvalRequested) coll.countSel(selector)
-          else fuccess(1000)
+      def nbResults: Fu[Int] = fuccess(1000)
 
-        def slice(offset: Int, length: Int): Fu[Seq[Streamer.WithUser]] =
-          coll
-            .aggregateList(length, readPreference = ReadPreference.secondaryPreferred) { framework =>
-              import framework.*
-              Match(selector) -> List(
-                Sort(if (approvalRequested) Ascending("updatedAt") else Descending("liveAt")),
-                Skip(offset),
-                Limit(length),
-                PipelineOperator(
-                  $lookup.simple(
-                    from = userRepo.coll,
-                    as = "user",
-                    local = "_id",
-                    foreign = "_id"
-                  )
-                ),
-                UnwindField("user")
+      def slice(offset: Int, length: Int): Fu[Seq[Streamer.WithContext]] =
+        coll
+          .aggregateList(length, readPreference = ReadPreference.secondaryPreferred) { framework =>
+            import framework._
+            Match(
+              $doc(
+                "approval.granted" -> true,
+                "listed"           -> Streamer.Listed(true),
+                "_id" $nin live.streams.map(_.streamer.id)
               )
-            }
-            .map { docs =>
-              for {
-                doc      <- docs
-                streamer <- doc.asOpt[Streamer]
-                user     <- doc.getAsOpt[User]("user")
-              } yield Streamer.WithUser(streamer, user)
-            }
-      }
-    )
+            ) -> List(
+              Sort(Descending("liveAt")),
+              Skip(offset),
+              Limit(length),
+              PipelineOperator(
+                $lookup.simple(
+                  from = userRepo.coll,
+                  as = "user",
+                  local = "_id",
+                  foreign = "_id"
+                )
+              ),
+              UnwindField("user"),
+              PipelineOperator(
+                $lookup.simple(
+                  from = subsRepo.coll,
+                  as = "subs",
+                  local = "_id",
+                  foreign = "s"
+                )
+              ),
+              AddFields($doc("subscribed" -> $doc("$in" -> List(forUser.??(_.toString), "$subs.u"))))
```

Comment:
it should be possible to avoid bringing all the subs in, for perf. By using $lookup.pipeline to only select the sub of `forUser`